Rest

AlexandrSid opened the pull request:
Сделан отдельный рест контроллер с монго для хранения текста статей отдельно от основной базы данных.
https://github.com/AlexandrSid/wikime_storage

Взаимодействие с ним со стороны wikime осуществляется через рализацию WebClient

AlexandrSid committed:
Dependencies and entry point of new functionality added.

AlexandrSid committed:
all examples realizations

AlexandrSid committed:
инструкция по запуску в fiasko.md

AlexandrSid committed:
инструкция по запуску в fiasko.md

telegram bot deleted

AlexandrSid committed:
get method finally works after .block(); added

AlexandrSid committed:
All REST methods works

AlexandrSid committed:
Text data store in separate mongo now, Mongo and it's REST controller are here: https://github.com/AlexandrSid/wikime_storage

AlexandrSid committed:
Update README.md

efremov-itcenter commented on src/main/java/org/aleksid/wikime/restclient/ArticleStorageRestClient.java:
```diff
@@ -0,0 +1,85 @@
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .bodyToMono(ArticleRestDao.class)
+                .block();
```
А нужна ли тут блокирующая операция? Может асинхронно работать с текстами статей? Хотя конечно, если возникнут какие-либо проблемы с сетью, например, можем нарваться на неконсистентность.

AlexandrSid replied:
Пока что-то новое не работает тяжело понять почему оно не работает. Сначала там был subscribe() как на гайдах по WebClient с baeldung.com и с ним ничего не работало, запросы на рест контроллер не приходили, даже get. После замены на block() get-запрос заработал. Потом был поиск рабочей схемы для отправки данных post put  методами уже в контексте работающего block(). Вообще возникает вопрос: а рест функциональность не надёжнее ли реализовать на основе kafka? тем более всё-равно это работает в облаках потом.

AlexandrSid committed:
всё ещё блокирующий способ

AlexandrSid committed:
Merge remote-tracking branch 'origin/rest' into rest

AlexandrSid committed:
Баг: при добавлении новых тегов к существующей статье, новым тегам при сохранении присваивается значение null, вплоть до repository.save() всё работает корректно вроде, новые статьи получают id-null и в таком виде идут на сохранение в составе статьи, в результате им присваивается id, а значение пропадает.

Больше не нужно вводить параграфы через запятую, просто текст. Модель оставлена прежняя

поле <input> текст страницы создания и редактирования статей увеличено до <textarea>

AlexandrSid committed:
Баг устранён путём сохранения всех новых тегов перед сохранением отредактированной статьи

TODO update

AlexandrSid committed:
Баг устранён путём сохранения всех новых тегов перед сохранением отредактированной статьи

AlexandrSid committed:
Merge remote-tracking branch 'origin/rest' into rest
